doc/man7/EVP_RAND.pod

```diff
@@ -201,15 +201,12 @@ entropy from a live entropy source (section 5.5.2 of [NIST SP 800-90C]).
 It is up to the user to ensure that a live entropy source is configured
 and is being used.
 
-For the three shared DRBGs (and only for these) there is another way to
-reseed them manually:
-If L<RAND_add(3)> is called with a positive I<randomness> argument
-(or L<RAND_seed(3)>), then this will immediately reseed the <primary> DRBG.
-The <public> and <private> DRBG will detect this on their next generate
-call and reseed, pulling randomness from <primary>.
-
-The last feature has been added to support the common practice used with
-previous OpenSSL versions to call RAND_add() before calling RAND_bytes().
+For the <primary> DRBG, calling L<RAND_add(3)> (or L<RAND_seed(3)>) will
+immediately reseed the DRBG.
+The <public> and <private> DRBGs will pick up the new randomness on
+their next reseeding.
+This supports the common practice, used with previous OpenSSL versions,
+to call L<RAND_add(3)> before calling L<RAND_bytes(3)> for the first time.
```

Comment:
I would put a little bit more emphasis on the changed behaviour, e.g. like this:

```suggestion
The <public> and <private> DRBGs will pick up the new randomness on
their next reseeding.
This supports the common practice, used with previous OpenSSL versions,
to call L<RAND_add(3)> before calling L<RAND_bytes(3)> for the first time.
(Note however, that contrary to earlier behaviour before version 3.2, the
<RAND_add(3)> call at application startup will not have an immediate effect
on the output of the following L<RAND_bytes(3)> call anymore, unless the
automatic initial seeding failed and the <public> DRBG is in an error state.) 
```

Reply:
This comes down to if we consider this a bug or a feature.  If it's a bug, back porting to 3.0 & 3.1 is acceptable.

Personally, I think it is a bug.  Others will likely have different views.

Reply:
The note is incorrect too I think.  A RAND_add call at startup will have an effect on the DRBGs.  A later call to RAND_add won't.

Reply:
You might be right. The <primary> DRBG will certainly seed before the application gets a chance to add randomness , unless you changed the behavior when you introduced providers: in 1.1.1, RAND_add() needs to obtain of the <primary>  DRBG in order to seed it by calling RAND_DRBG_get0_primary(), which inevitably triggers the initial seeding. The thread-local instances, however, might still be unseeded.

Reply:
That's what I thought would happen.  The primary would initialise and seed but the thread local instances would be initialised later.

I've tweaked the wording but it's probably not ideal yet.

Reply:
Your rewording looks ok to me.